sup3r/models/multi_step.py

```diff
@@ -131,6 +112,82 @@ def seed(s=0):
         """
         Sup3rGan.seed(s=s)
 
+    def _get_model_step_exo(self, model_step, exogenous_data=None):
+        """Get the exogenous data for the given model_step from the full
+        exogenous data dictionary
+
+        Parameters
+        ----------
+        model_step : int
+            Index of the model to get exogenous data for.
+        exogenous_data : dict
+            Dictionary of exogenous feature data with entries describing
+            whether features should be combined at input, a mid network layer,
+            or with output. e.g.
+            {'topography': {'steps': [
+                {'combine_type': 'input', 'model': 0, 'data': ...,
+                 'resolution': ...},
+                {'combine_type': 'layer', 'model': 0, 'data': ...,
+                 'resolution': ...}]}}
+            Each array in in 'data' key has 3D or 4D shape:
+            (spatial_1, spatial_2, 1)
+            (spatial_1, spatial_2, n_temporal, 1)
+
+        Returns
+        -------
+        exogenous_data : dict
+            Same as input dictionary but with only entries with 'model':
+            model_step
+        """
+        model_step_exo = None
+        if exogenous_data is not None:
+            model_step_exo = {}
+            for feature in exogenous_data:
+                steps = [step for step in exogenous_data[feature]['steps']
+                         if step['model'] == model_step]
+                if steps:
+                    model_step_exo[feature] = {'steps': steps}
```

Comment:
It might be worth defining a separate data structure for the `exogenous_data` dictionary, especially if it defines the data structure more explicitly thereby helping users decipher how to set up the config. 

Consider a couple of classes: `class SingleExoDataStep(combine_type, model, data)`, `class ExoData(steps)`... Would be extra good if `ExoData` could be init from a config file directly OR a collection of `SingleExoDataStep` instances. Then the `ExoData` class would have methods to retrieve various bits of the data. 

Not totally sure how this would fit into the refactored fwp config as i haven't gotten there yet.

Reply:
This is a cool idea. Maybe better to leave for another PR so this initial big refactor can be merged first?

Reply:
Yeah that's fine, i think separate data struct classes and removing the spatial/temporal model are the two pieces left "todo", right?

Reply:
Just the struct classes, if were ok with keeping the SpatialThenTemporalBase used only by the Solar model.